### PR TITLE
Prepare app for docker

### PR DESCRIPTION
**Pull request by pwyborn**
@colinbruce could you please just check my changes (as we discussed yesterday). I have actually deployed this branch to demo and staging so you can check. It appears to be working ok (for the existing app - I will have to revisit the docker version as I am getting some problems with the load balancer - don't think this is app related though).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,10 +16,11 @@ ENV \
     ANONYMOUS_PLACEHOLDER_EMAIL="" \
     RAILS_RELATIVE_URL_ROOT='' \
     PDFTK="/usr/bin/pdftk" \
- #  REDIS_STORE="redis://localhost:6379/1" \
+    REDIS_STORE="redis://localhost:6379/1" \
  #  RAILS_ENV="development" \
- #  ENV_NAME="development" \
-    GA_ID=""
+    ENV_NAME="development" \
```
Comment (colinbruce) [inline Dockerfile]: I'm not sure we should have development on the production dockerfile.

File: Dockerfile [hunk 2]
```diff
@@ -49,12 +50,12 @@ COPY ./docker/runit/runit-service /etc/service/accelerated_claims/run
 RUN bundle exec rake assets:precompile RAILS_ENV=production SECRET_TOKEN=blah
```
Comment (colinbruce) [inline Dockerfile]: Because here we set it to production anyway?! :/

File: Dockerfile [hunk 3]
```diff
@@ -49,12 +50,12 @@ COPY ./docker/runit/runit-service /etc/service/accelerated_claims/run
...
 
 RUN chmod +x /etc/service/strike2/run /etc/service/accelerated_claims/run
-#RUN apt-get install -y redis-serve
 RUN apt-get install -y vim
-# RUN service redis-server start
 RUN cd /usr/src/app/log
 RUN chmod 777 *
 
+
```
Comment (colinbruce) [inline Dockerfile]: Two lines of whitespace! :)

File: config/environments/production.rb [hunk 4]
```diff
@@ -90,6 +94,12 @@
   # Use default logging formatter so that PID and timestamp are not suppressed.
   config.log_formatter = ::Logger::Formatter.new
 
-  # Session store (cache_store) - using elasticache endpoints url
-  config.cache_store = :redis_store, (ENV['REDIS_URL'])
+  if ENV['DOCKER_ENV'].present?
+    # Session store (cache_store) - using elasticache endpoints url
+    config.cache_store = :redis_store, (ENV['REDIS_URL'] || 'redis://localhost:6379/1')
```
Comment (colinbruce) [inline config/environments/production.rb]: If docker_env is set do we need the optional fallback of `'redis://localhost:6379/1'`, I thought that in docker localhost would never work?

**Comment by pwyborn** [comment]
@colinbruce Hopefully I have corrected all of your points thanks

**Comment by colinbruce** [comment]
This looks fine, lets merge and test :+1:

**Comment by pwyborn** [comment]
@colinbruce - as discussed ENV_NAME changed to "". All appears to be well in staging and demo (civil-claims). Also cache_store correctly set to ""  in docker version

**Pull request merged**